File: Dataset/Leetcode/train/110/53.java
class Solution {
    public boolean XXX(TreeNode root) {
        return root == null || XXX(root.left) && XXX(root.right) && (Math.abs(depth(root.left) - depth(root.right)) <= 1);
    }

    //求树的深度
    public static int depth(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.max(depth(root.left), depth(root.right));
    }
}

File: Dataset/Leetcode/train/70/53.c
int XXX(int n){
    if(n==1)
       return 1;
    else if(n==2)
       return 2;
    int a=1,b=2;
    for(int i=2;i<n;i++){
        b = a + b;
        a = b - a;
    }
    return b;
}

